Sources/Testing/Events/Event.swift

```diff
@@ -168,6 +168,28 @@ public struct Event: Sendable {
   /// The instant at which the event occurred.
   public var instant: Test.Clock.Instant
 
+  /// Storage for ``sourceLocation``.
+  private var _sourceLocation: SourceLocation?
```

Comment:
I'm a little wary of the perf implications of this additional storage on `Event`.

I see you propagated the new `sourceLocation:` parameter on `.expectationChecked` events, but the `Expectation` type already has a non-Optional source location property, so I think there should already be a value for those particular events to use.

I'm wondering if we really need this, or could we instead propagate the source location of an attachment via the payload of the `.valueAttachmentFailed` enum case? Alternatively, perhaps this could be a new property on `Event.Context`?

Reply:
Actually, I can just go ahead and put the storage back on `Attachment` and remove it from `Event`.